lib/Dialect/SPIRV/SPIRVOps.cpp

```diff
@@ -1053,6 +1053,67 @@ static LogicalResult verify(spirv::BranchConditionalOp branchOp) {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// spv.CompositeConstruct
+//===----------------------------------------------------------------------===//
+
+static ParseResult parseCompositeConstructOp(OpAsmParser &parser,
+                                             OperationState &state) {
+  SmallVector<OpAsmParser::OperandType, 4> operands;
+  Type type;
+  auto loc = parser.getCurrentLocation();
+
+  if (parser.parseOperandList(operands) || parser.parseColonType(type)) {
+    return failure();
+  }
+  auto cType = type.dyn_cast<spirv::CompositeType>();
+  if (operands.size() != cType.getNumElements()) {
+    return parser.emitError(loc, "has incorrect number of operands: expected ")
+           << cType.getNumElements() << ", but provided " << operands.size();
+  }
+  // TODO: Add support for constructing a vector type from the vector operands.
+  // According to the spec: "for constructing a vector, the operands may
+  // also be vectors with the same component type as the Result Type component
+  // type".
+  SmallVector<Type, 4> elementTypes;
+  for (uint32_t i = 0, e = cType.getNumElements(); i < e; ++i) {
+    elementTypes.push_back(cType.getElementType(i));
+  }
+  return failure(
+      parser.resolveOperands(operands, elementTypes, loc, state.operands) ||
+      parser.addTypesToList(type, state.types));
```

Comment:
Swap the order of these:

state.addTypes(elementTypes);
return parser.resolveOperands(...);

`addTypesToList` always returns success.

Reply:
thanks, fixed.